eclipse-che/che-server
# fix: devfile endpoint with single-host exposure

…devfile endpoint exposure is enabled

Signed-off-by: Michal Vala <mvala@redhat.com>



### What does this PR do?
See analysis of https://github.com/eclipse/che/issues/20593

This PR basically checks the `CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_DEVFILE__ENDPOINT__EXPOSURE` and set `requireSubdomain` only for cases when it is not `single-host`.


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20593

### How to test this PR?


latest image `quay.io/mvala/che-server:gh20593-singleHostDevfileEndpoint`

1. deploy with `chectl server:deploy --platform openshift --cheimage=quay.io/mvala/che-server:gh20593-singleHostDevfileEndpoint --che-operator-cr-patch-yaml=patch.yml` where patch.yml is:
```
spec:
  server:
    customCheProperties:
      CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_DEVFILE__ENDPOINT__EXPOSURE: single-host
    serverExposureStrategy: single-host
```
2. Start any workspace that has endpoint defined in the devfile component
3. Workspace must start and be accessible from the browser (prone to fail is .net core sample devfile, but it's not failing 100%)
4. Run workspace task to use devfile endpoint and open it from theia, it must open on correct url and must work.
5. You can also observe browser network traffic for `GET /workspace` request, it must have server for the devfile endpoint with proper path suffix, like `/`


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
